revolutionpi/revolution_pi.go

```diff
@@ -145,13 +149,40 @@ func (b *revolutionPiBoard) DoCommand(ctx context.Context,
 ) (map[string]interface{}, error) {
 	resp := make(map[string]interface{})
 
-	_, ok := req["getStatus"]
-	if ok {
-		err := b.controlChip.showDeviceList()
+	pinMessage, exists := req[readAddressKey]
+	if exists {
+		pinName, ok := pinMessage.(string)
+		if !ok {
+			return nil, fmt.Errorf("error performing readAddress: expected string got %v", pinMessage)
+		}
+		pin := SPIVariable{strVarName: char32(pinName)}
+		err := b.controlChip.mapNameToAddress(&pin)
 		if err != nil {
 			return nil, err
 		}
-		resp["getStatus"] = "status ok"
+		b.controlChip.logger.Debugf("pin: %#v", pin)
```

Comment:
Mild preference to make the logging statement a little more explicit. Maybe "reading pin %#v" or something, so we know why we're logging a thing about the pin. 

...maybe my suggestion isn't actually needed, because the logging should include the current line number and from that you can look up what's going on in the code. Something to think about.